http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/resources/FileUploadResource.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.ts.http.restclient.reactive.resources;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.stream.Collectors;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import io.quarkus.ts.http.restclient.reactive.multipart.MultipartBodyWithMediaType;
+import io.quarkus.ts.http.restclient.reactive.multipart.MultipartBodyWithoutMediaType;
+
+@Path("/upload")
+public class FileUploadResource {
+    @POST
+    @Path("/without")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    public String uploadFileWithoutMediaType(MultipartBodyWithoutMediaType data) {
+        String content = new BufferedReader(new InputStreamReader(data.file))
```

Comment:
I am not sure you are right. I inspected `BufferedReader` code and there is nothing that indicates it can close itself when all the lines are read. I take it you believe that you don't need to close resources because Quarkus provided IS, so it is going to close it?

If so, could you indicate some docs that says that, or have you tried it? I suppose you can add store that IS from one call in a volatile var and check with a call to different endpoint if that IS is closed or something.

I don't think this is super important issue, I am just curious how have you come to our conclusions. Thanks

Reply:
From my POV it should be closed. What I know is that the resources are closed from top so if Quarkus close `MultipartBodyWithoutMediaType data` input stream in background. I think that the `BufferedReader` will stay open.